[DX-1726] Dashboard release notes 5.3.8 (#5659)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-dashboard/release-notes/version-5.3.md

As outlined in the [LTS policy]({{< ref "developer-support/special-releases-and-features/long-term-support-releases" >}}), version 5.3 is a long-term support release with full support available until May 2025. Maintenance support will continue until May 2026. Our next long-term support release will be announced at the end of April 2025.

---

## 5.3.8 Release Notes

### Release Date xxx

### Breaking Changes

> There are no breaking changes in this release, however, if moving from a version of Tyk older than 5.3.0 please read the explanation provided with [5.3.0 release]({{< ref "#TykOAS-v5.3.0">}}).

Reviewer: Please follow the [RN](https://github.com/TykTechnologies/tyk-docs/blob/master/tyk-docs/assets/others/release-notes-template.md#breaking-changes) format

### Deprecations

> This is an advanced notice that External OAuth, OpenID Connect (OIDC) authentication methods, and SQLite support will be deprecated starting in version 5.7.0. Please review your API configurations, as the Gateway logs will provide notifications for any APIs utilizing these methods.

Reviewer: Please add relevant reference links to External OAuth, OpenID Connect, Sqlite

Author: We haven't created a deprecation log yet so we don't have one single reference page that has all the info in. We have placed warning messages on all pages in the docs that have these authentication methods mentioned.

### Upgrade Instructions

> If you are upgrading to 5.3.8, please follow the detailed [upgrade instructions](#upgrading-tyk).

Reviewer: Please follow the [RN](https://github.com/TykTechnologies/tyk-docs/blob/master/tyk-docs/assets/others/release-notes-template.md#upgrading-tyk) format

### Release Highlights
This release focuses mainly on bug fixes. For a comprehensive list of changes, please refer to the detailed [changelog]({{< ref "#Changelog-v5.3.8">}}) below.

### Dependencies {#dependencies-5.3.8}

#### Compatibility Matrix For Tyk Components
<!-- Required. Version compatibility with other components in the Tyk stack. This takes the form of a compatibility matrix and is only required for Gateway and Portal.
An illustrative example is shown below. -->
| Dashboard Version | Recommended Releases | Backwards Compatibility |
|----    |---- |---- |
| 5.3.8 | MDCB v2.5.1     | MDCB v2.5.1 |
|         | Operator v0.17 | Operator v0.16 |
|         | Sync v1.4.3   | Sync v1.4.3 |

> |         | Helm Chart (tyk-stack, tyk-oss, tyk-dashboard, tyk-gateway) v1.4.0 | Helm all versions |

Reviewer: Shouldn't the version for helm be `v2`?

| | EDP v1.8.3 | EDP all versions |
| | Pump v1.9.0 | Pump all versions |
| | TIB (if using standalone) v1.5.1 | TIB all versions |

#### 3rd Party Dependencies & Tools {#3rdPartyTools-v5.3.8}
<!-- Required. Third-party dependencies encompass tools (GoLang, Helm etc.), databases (PostgreSQL, MongoDB etc.) and external software libraries. This section should be a table that presents the third-party dependencies and tools compatible with the release. Compatible is used in the sense of those versions tested with the releases. Such information assists customers considering upgrading to a specific release.
Additionally, a disclaimer statement was added below the table, for customers to check that the third-party dependency they decide to install remains in support.
An example is given below for illustrative purposes only. Tested Versions and Compatible Versions information will require discussion with relevant squads and QA. -->

| Third Party Dependency                                     | Tested Versions        | Compatible Versions    | Comments |
| ---------------------------------------------------------- | ---------------------- | ---------------------- | -------- |

> | [GoLang](https://go.dev/dl/)                               | 1.21       | 1.21       | [Go plugins]({{< ref "plugins/supported-languages/golang" >}}) must be built using Go 1.21 |

Reviewer: This should be `1.22`? i guess. Please check with previous release.

Please check both the tables with previous release & If necessary confirm from @caroltyk

| [Redis](https://redis.io/download/)  | 6.2.x, 7.x  | 6.2.x, 7.x  | Used by Tyk Dashboard |
| [MongoDB](https://www.mongodb.com/try/download/community)  | 5.0.x, 6.0.x, 7.0.x  | 5.0.x, 6.0.x, 7.0.x  | Used by Tyk Dashboard |
| [PostgreSQL](https://www.postgresql.org/download/)         | 12.x - 16.x LTS        | 12.x - 16.x            | Used by Tyk Dashboard |
| [OpenAPI Specification](https://spec.openapis.org/oas/v3.0.3) | v3.0.x      | v3.0.x          | Supported by [Tyk OAS]({{< ref "tyk-apis/tyk-gateway-api/oas/x-tyk-oas-doc" >}})|

### Downloads
- [Docker Image to pull](https://hub.docker.com/r/tykio/tyk-dashboard/tags?page=&page_size=&ordering=&name=v5.3.8)

>  - ```bash

Reviewer: This is not rendered as a sub point

Reviewer: This is not resolved yet.

   docker pull tykio/tyk-dashboard:v5.3.8
   ```
- Helm charts

>  - [tyk-charts v1.4]({{< ref "product-stack/tyk-charts/release-notes/version-1.4.md" >}})

Reviewer: This is not rendered as a sub point

>  - [tyk-charts v1.4]({{< ref "product-stack/tyk-charts/release-notes/version-1.4.md" >}})

Reviewer: This should also be `v2`

Reviewer: This is also not resolved yet.

#### Fixed
<!-- This section should be a bullet point list of new features. Explain:

- The purpose of the new feature
- How does the new feature benefit users?
- Link to documentation of the new feature
- For OSS - Link to the corresponding issue if possible on GitHub to allow the users to see further info.

Each change log item should be expandable. The first line summarises the changelog entry. It should be then possible to expand this to reveal further details about the changelog item. This is achieved using HTML as shown in the example below. -->
<ul>
<li>
<details>
<summary>User Group dropdown limitations in Dashboard</summary>

This update resolves an issue with the user group dropdown, ensuring that all created user groups are displayed when creating a new user.
</details>
</li>
<li>
<details>
<summary>Rate Limiting not saved with Upstream Certificates enabled for OAS API</summary>

Fixed an issue in the Tyk OAS API Designer where Rate Limiting settings were not saved when Upstream Certificates were enabled. This fix ensures that both Rate Limits and Upstream Certificates configurations can now be saved together
</details>
</li>
<li>
<details>

> <summary>Deprecation notice of External OAuth and OpenID Connect options</summary>

Reviewer: I think this should not come in the `fixed` section. Used [added](https://github.com/TykTechnologies/tyk-docs/blob/master/tyk-docs/assets/others/release-notes-template.md#added) instead.

> A deprecation notice for External OAuth and OpenID Connect (OIDC) authentication mechanisms has been implemented in the UI starting from version 5.3.8. This provides advanced notification to users regarding any APIs configured with these authentication methods in preparation for future upgrades where these middleware options may be removed in version 5.7.0.

Reviewer: Add a reference to the notice.

| ---------------------------------------------------------- | ---------------------- | ---------------------- | -------- |
| [GoLang](https://go.dev/dl/)                               | 1.22       | 1.22       | [Go plugins]({{< ref "plugins/supported-languages/golang" >}}) must be built using Go 1.22 |
| [Redis](https://redis.io/download/)  | 6.2.x, 7.x  | 6.2.x, 7.x  | Used by Tyk Dashboard |
| [MongoDB](https://www.mongodb.com/try/download/community)  | 5.0.x, 6.0.x, 7.0.x  | 5.0.x, 6.0.x, 7.0.x  | Used by Tyk Dashboard |
| [PostgreSQL](https://www.postgresql.org/download/)         | 12.x - 16.x LTS        | 12.x - 16.x            | Used by Tyk Dashboard |
| [OpenAPI Specification](https://spec.openapis.org/oas/v3.0.3) | v3.0.x      | v3.0.x          | Supported by [Tyk OAS]({{< ref "tyk-apis/tyk-gateway-api/oas/x-tyk-oas-doc" >}})|

Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

### Deprecations

> This is an advanced notice that External OAuth, OpenID Connect (OIDC) authentication methods, and SQLite support will be deprecated starting in version 5.7.0. Please review your API configurations, as the Gateway logs will provide notifications for any APIs utilizing these methods.

Reviewer 2: ```suggestion
This is an advanced notice that the dedicated External OAuth, OpenID Connect (OIDC) authentication options, and SQLite support will be deprecated starting in version 5.7.0. We recommend that users of the [External OAuth]({{< ref "basic-config-and-security/security/authentication-authorization/ext-oauth-middleware" >}}) and [OpenID Connect]({{< ref "basic-config-and-security/security/authentication-authorization/openid-connect" >}}) methods migrate to Tyk's dedicated [JWT Auth]({{< ref "basic-config-and-security/security/authentication-authorization/json-web-tokens#about-jwts" >}}) method. Please review your API configurations, as the Gateway logs will provide notifications for any APIs utilizing these methods.
```

I'm thinking we should provide reassurance here that there is a recommended alternative (JWT Auth). WDYT?

### Upgrade Instructions
If you are upgrading to 5.3.8, please follow the detailed [upgrade instructions](#upgrading-tyk).

### Downloads
- [Docker Image to pull](https://hub.docker.com/r/tykio/tyk-dashboard/tags?page=&page_size=&ordering=&name=v5.3.8)
 - ```bash
   docker pull tykio/tyk-dashboard:v5.3.8
   ```
- Helm charts
 - [tyk-charts v1.4]({{< ref "product-stack/tyk-charts/release-notes/version-1.4.md" >}})

### Changelog {#Changelog-v5.3.8}
<!-- Required. The change log should include the following ordered set of sections below that briefly summarise the features, updates and fixed issues of the release.
Here it is important to explain the benefit of each changelog item. As mentioned by James in a previous Slack message (https://tyktech.slack.com/archives/C044R3ZTN6L/p1686812207060839?thread_ts=1686762128.651249&cid=C044R3ZTN6L):
"...it is important to document the customer impact for the work delivered, so we can share it with prospects/install base. For example:
"New Chart delivers x and y benefit to a and b customer use cases. The business impact for them will be this and that" -->
#### Added

<ul>
<li>
<details>

> <summary>Deprecation notice of External OAuth and OpenID Connect options</summary>

Reviewer 2: ```suggestion
<summary>Advanced notice of deprecation of dedicated External OAuth and OpenID Connect auth options</summary>
```

Trying to soften the blow a bit here - don't want to scare users into thinking that they can no longer use OAuth or OIDC.

> A deprecation notice for External OAuth and OpenID Connect (OIDC) authentication mechanisms has been implemented in the UI starting from version 5.3.8. This provides advanced notification to users regarding any APIs configured with these authentication methods in preparation for future upgrades where these middleware options may be removed in version 5.7.0.

Reviewer 2: ```suggestion
The UI now displays a deprecation notice for the dedicated [External OAuth]({{< ref "basic-config-and-security/security/authentication-authorization/ext-oauth-middleware" >}}) and [OpenID Connect (OIDC)]({{< ref "basic-config-and-security/security/authentication-authorization/openid-connect" >}}) authentication mechanisms. This provides advanced notification that these authentication options will be deprecated in version 5.7.0. Users are advised to migrate to the [JWT Auth]({{< ref "basic-config-and-security/security/authentication-authorization/json-web-tokens#about-jwts" >}}) method, which supports integration with both OAuth and OIDC providers, in preparation for future upgrade.
```

Proposed rewording with links into the docs.

</details>
</li>
</ul>

#### Fixed
<!-- This section should be a bullet point list of new features. Explain:

- The purpose of the new feature
- How does the new feature benefit users?
- Link to documentation of the new feature
- For OSS - Link to the corresponding issue if possible on GitHub to allow the users to see further info.

Each change log item should be expandable. The first line summarises the changelog entry. It should be then possible to expand this to reveal further details about the changelog item. This is achieved using HTML as shown in the example below. -->
<ul>
<li>
<details>
<summary>User Group dropdown limitations in Dashboard</summary>

> This update resolves an issue with the user group dropdown, ensuring that all created user groups are displayed when creating a new user.

Reviewer 2: ```suggestion
Fixed an issue with the user group dropdown in the Dashboard UI, ensuring that all available user groups are displayed when creating a new user.
```

</details>
</li>
<li>
<details>

> <summary>Rate Limiting not saved with Upstream Certificates enabled for OAS API</summary>

Reviewer 2: ```suggestion
<summary>Rate Limiting settings not saved when Upstream Certificates enabled for Tyk OAS API</summary>
```

Please remember always to use "Tyk OAS API" never abbreviate to "OAS API" as that can lead to uncertainty/confusion.
